core/options.go

```diff
@@ -0,0 +1,27 @@
+package core
```

Comment:
I would not create an options func here. This style is great don't get me wrong, but given Changie is never imported there isn't really a need for option funcs. We can just add the parameter to the functions that need it. If there are too many parameters, like the batch function, then we can create a parameter struct.

Reply:
I believe it is your call to decide to break the API of the core package. I will adapt if you add a breaking change there.

I don't want to argue about how we pass options. Function vs struct, either is fine by me. Since there was none, I've added the way I personally favor.
 
My intent with the option functor pattern was the following:
* impact on existing code (including tests) with this option has the smallest footprint. It is easy to review and test.
* it doesn't break the API of the core package
* you don't know whether `core` is never imported: since this package is not internal, it is exported and therefore potentially imported by other tools. For instance, I could decide to build my own `changie` CLI based on this package
* since what I like with `changie` is flexibility (over say, git commit-based changelog solutions), I believe there is a great potential for allowing a tuned behavior (see my other change proposal with a --force-patch flag)

Reply:
Not done. Pending your decision to change the core package API

Reply:
In the readme I specify that the breaking changes rules applies to the cli commands and configs and that internally the code may have breaking changes. There is no internal folder as it is the entire project.